test/components/legendTests.ts

```diff
@@ -254,6 +254,34 @@ describe("Legend", () => {
     svg.remove();
   });
 
+  it("accept formatter and show entity labels correcly", () => {
+    color.domain(["jtao", "mschafer", "kfalter"]);
+    let data = [
+      {id: "jtao", name: "Joy Tao", score: 13},
+      {id: "mschafer", name: "Mark Schafer", score: 15},
+      {id: "kfalter", name: "Kelsey Falter", score: 11}
+    ];
+    let f = (id: string) => {
+      let pickedEntity = data.filter(function(d) {return (d.id === id); });
+      if (pickedEntity.length > 0) {
+          return pickedEntity[0].name;
+        }else {
+          return "NA";
+        }
+    };
+    legend.formatter(f);
+    legend.renderTo(svg);
+
+    let texts = svg.selectAll("title")[0].map(function(text: Element) {return text.textContent; });
```

Comment:
I believe D3 has a `text()` API that can be incredibly useful.

https://github.com/mbostock/d3/wiki/Selections#text